### PR TITLE
Integrate vector to liegroups

### PR DESCRIPTION
**Pull request by cjpurackal**
#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/geometry/liegroup/se3.py [hunk 1]
```diff
@@ -37,18 +24,19 @@ class Se3(Module):
...
         >>> s.t
-        Parameter containing:
-        tensor([1., 1., 1.], requires_grad=True)
+        x: 1.0
+        y: 1.0
+        z: 1.0
     """
 
-    def __init__(self, rotation: So3, translation: Tensor) -> None:
+    def __init__(self, rotation: So3, translation: Union[Vector3, Tensor]) -> None:
```
Comment (cjpurackal) [inline kornia/geometry/liegroup/se3.py:32]: Should we only allow for Vector3 here? @edgarriba

**Review by edgarriba — APPROVED** [review]
LGTM

File: kornia/geometry/liegroup/se2.py [hunk 2]
```diff
@@ -61,9 +62,15 @@ def __init__(self, rotation: So2, translation: Tensor) -> None:
         super().__init__()
         KORNIA_CHECK_TYPE(rotation, So2)
         # TODO change to KORNIA_CHECK_SHAPE once there is multiple shape support
-        check_se2_r_t_shape(rotation, translation)
-        self._rotation = rotation
-        self._translation = Parameter(translation)
+        # KORNIA_CHECK_TYPE(translation, (Vector3, Tensor))
+        assert isinstance(translation, (Vector2, Tensor)), f"translation type is {type(translation)}"
```
[the remark this answers is not in the document]
Comment (cjpurackal) [reply kornia/geometry/liegroup/se2.py:66]: yea, I will try to make a separate PR for that.

File: kornia/geometry/liegroup/se3.py [hunk 3]
```diff
@@ -63,17 +63,23 @@ def __init__(self, rotation: So3, translation: Tensor) -> None:
...
+            self._translation = translation
 
     def __repr__(self) -> str:
         return f"rotation: {self.r}\ntranslation: {self.t}"
 
     def __getitem__(self, idx: Union[int, slice]) -> 'Se3':
         return Se3(self._rotation[idx], self._translation[idx])
 
-    def __mul__(self, right: "Se3") -> "Se3":
+    def __mul__(self, right: "Se3") -> Union['Se3', Vector3, Tensor]:
```
Comment (johnnv1) [inline kornia/geometry/liegroup/se3.py:82]: any case which block us of having `from __future__ import annotations` here too?

File: kornia/geometry/liegroup/so3.py [hunk 4]
```diff
@@ -6,6 +6,7 @@
 from kornia.core.check import KORNIA_CHECK_TYPE
 from kornia.geometry.linalg import batched_dot_product
 from kornia.geometry.quaternion import Quaternion
+from kornia.geometry.vector import Vector3
```
Comment (johnnv1) [inline kornia/geometry/liegroup/so3.py:9]: if possible, add `from future import annotations` here too

**Review by johnnv1 — APPROVED** [review]
LGTM, as soon the pre-commit issues for this PR are fixed

File: kornia/geometry/liegroup/se2.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnnv1) [reply kornia/geometry/liegroup/se2.py:68]: #2353

File: kornia/geometry/liegroup/se2.py [hunk 5]
```diff
@@ -61,18 +64,24 @@ def __init__(self, rotation: So2, translation: Tensor) -> None:
...
```
Comment (johnnv1) [inline kornia/geometry/liegroup/se2.py:68]:
```suggestion
        if not isinstance(translation, (Vector2, Tensor)):
            raise TypeError(f"translation type is {type(translation)}")
```

File: kornia/geometry/liegroup/se3.py [hunk 6]
```diff
@@ -63,17 +63,23 @@ def __init__(self, rotation: So3, translation: Tensor) -> None:
         """
         super().__init__()
         KORNIA_CHECK_TYPE(rotation, So3)
+        # KORNIA_CHECK_TYPE(translation, (Vector3, Tensor))
+        assert isinstance(translation, (Vector3, Tensor)), f"translation type is {type(translation)}"
```
Comment (johnnv1) [inline kornia/geometry/liegroup/se3.py:67]:
```suggestion
        if not isinstance(translation, (Vector3, Tensor)):
            raise TypeError(f"translation type is {type(translation)}")
```

**Pull request merged**